File: labor1/app/src/main/java/com/example/labor1/registerActivity.java
package com.example.labor1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class registerActivity extends AppCompatActivity {

    private static int callFromProfile = 0;
    private static int callFromLogin = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            String caller = bundle.getString("caller");
            if (Objects.equals(caller, "profile")) {
                callFromProfile++;
            } else if (Objects.equals(caller, "login")) {
                callFromLogin++;
            }
        }
        ((TextView) this.findViewById(R.id.profileCallCounter)).setText(String.valueOf(callFromProfile));
        ((TextView) this.findViewById(R.id.loginCallCounter)).setText(String.valueOf(callFromLogin));
    }

    public void callProfile(View view) {
        Intent profileActivity = new Intent(this, ProfileActivity.class)
                .putExtra("firstName", ((EditText) this.findViewById(R.id.firstNameText)).getText().toString())
                .putExtra("lastName", ((EditText) this.findViewById(R.id.lastNameText)).getText().toString())
                .putExtra("department", ((EditText) this.findViewById(R.id.departmentText)).getText().toString())
                .putExtra("caller", "register");
        startActivity(profileActivity);
    }
}

File: labor1/app/src/main/java/com/example/labor1/RegisterActivity.kt
package com.example.labor1;

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class RegisterActivity : AppCompatActivity() {

    companion object {
        @JvmStatic
        var callFromProfile = 0
        @JvmStatic
        var callFromLogin = 0
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val bundle = intent.extras
        if (bundle != null) {
            val caller = bundle.getString("caller")
            if (Objects.equals(caller, "profile")) {
                callFromProfile++
            } else if (Objects.equals(caller, "login")) {
                callFromLogin++
            }
        }
        findViewById<TextView>(R.id.profileCallCounter).text = callFromProfile.toString()
        findViewById<TextView>(R.id.loginCallCounter).text = callFromLogin.toString()
    }

    fun callProfile(view: View) {
        val profileActivity = Intent(this, ProfileActivity::class.java)
                .putExtra("firstName", this.findViewById<EditText>(R.id.firstNameText).text.toString())
                .putExtra("lastName", findViewById<EditText>(R.id.lastNameText).text.toString())
                .putExtra("department", findViewById<EditText>(R.id.departmentText).text.toString())
                .putExtra("caller", "register")
        startActivity(profileActivity)
    }
}
